File: app/src/main/java/com/hencoder/hencoderpracticedraw2/practice/Practice02RadialGradientView.java
package com.hencoder.hencoderpracticedraw2.practice;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.RadialGradient;
import android.graphics.Shader;
import android.graphics.SweepGradient;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;

public class Practice02RadialGradientView extends View {

    private Paint mPaint;

    public Practice02RadialGradientView(Context context) {
        this(context, null);
    }

    public Practice02RadialGradientView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public Practice02RadialGradientView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public void init() {
        mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        Shader shader = new RadialGradient(300, 300, 200, 0XFFE91E63, 0XFF2196F3, Shader.TileMode.REPEAT);
        mPaint.setShader(shader);
        // 用 Paint.setShader(shader) 设置一个 RadialGradient
        // RadialGradient 的参数：圆心坐标：(300, 300)；半径：200；颜色：#E91E63 到 #2196F3
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);

        canvas.drawCircle(300, 300, 200, mPaint);
    }

}

File: app/src/main/java/com/hencoder/hencoderpracticedraw2/practice/Practice02RadialGradientView.kt
package com.hencoder.hencoderpracticedraw2.practice

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.RadialGradient
import android.graphics.Shader
import android.util.AttributeSet
import android.view.View

/**
 * Created by xh on 2017/10/17.
 */
class Practice02RadialGradientView : View {

    private val mPaint = Paint(Paint.ANTI_ALIAS_FLAG)
    private var mRadiaGradient: RadialGradient

    private val mRadius = 200f

    private val mStartColor = 0XFFE91E63.toInt()
    private val mEndColor = 0XFF2196F3.toInt()

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    init {
        mRadiaGradient = RadialGradient(300f, 300f, mRadius, mStartColor, mEndColor, Shader.TileMode.CLAMP)
        mPaint.shader = mRadiaGradient
    }

    override fun onDraw(canvas: Canvas?) {
        super.onDraw(canvas)

        canvas!!.drawCircle(300f, 300f, mRadius, mPaint)

    }

}